tests/foreman/ui_airgun/test_smartvariable.py

```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+
+    :id: e98a8404-5e32-4d2e-af81-4339d214658a
+
+    :steps:
+
+        1.  Create a variable.
+        2.  Associate variable with fqdn/hostgroup.
+        3.  Create a matcher for fqdn/hostgroup with valid details.
+        4.  From host/hostgroup, edit the variable value.
+        5.  Submit the changes.
+
+    :expectedresults:
+
+        1.  The host/hostgroup is saved with changes.
+        2.  Matcher value in variable is updated from fqdn/hostgroup.
+
+    :CaseLevel: Integration
+    """
+    variable_name = gen_string('alpha')
+    variable_value = gen_string('numeric').lstrip('0')
+    variable_matcher_value = gen_string('numeric').lstrip('0')
+    variable_new_value = gen_string('numeric').lstrip('0')
+    with session:
+        session.smartvariable.create({
+            'variable.key': variable_name,
+            'variable.puppet_class': puppet_class.name,
+            'variable.key_type': 'integer',
+            'variable.default_value': variable_value,
+            'variable.matchers': [
+                {
+                    'Attribute type': {
+                        'matcher_attribute_type': 'fqdn',
+                        'matcher_attribute_value': module_host.name
+                    },
+                    'Value': variable_matcher_value
+                },
+            ]
+        })
+        values = session.smartvariable.search(variable_name)
+        assert values[0]['Variable'] == variable_name
+        # note that we should not enforce override, the variable should already be displayed as
+        # overridden
+        session.host.set_puppet_class_parameter_value(
+            module_host.name,
+            variable_name,
+            dict(value=variable_new_value)
+        )
+        host_smart_variable_value = session.host.get_puppet_class_parameter_value(
+            module_host.name, variable_name)
+        assert host_smart_variable_value['value'] == variable_new_value
+        values = session.smartvariable.read(variable_name)
+        assert len(values['variable']['matchers']['table']) == 1
+        property_matcher = values['variable']['matchers']['table'][0]
+        assert property_matcher['Attribute type']['matcher_attribute_type'] == 'fqdn'
+        assert property_matcher['Attribute type']['matcher_attribute_value'] == module_host.name
+        assert property_matcher['Value']['value'] == variable_new_value
+
+
+@tier2
+def test_negative_update_matcher_from_attribute(session, module_host, puppet_class):
+    """No impact on variable on editing the variable with invalid value
+    from attribute.
+
+    :id: bd4a2535-57dd-49a8-b8b5-c5e8de652aa7
+
+    :steps:
+
+        1.  Create a variable.
+        2.  Associate variable with fqdn/hostgroup.
+        3.  Create a matcher for fqdn/hostgroup with valid details.
+        4.  From host/hostgroup, attempt to edit the variable with invalid
+            value.
+
+    :expectedresults:
+
+        1.  Error thrown for invalid value.
```

Comment:
same here. We need to check that proper error is shown on UI

Reply:
addressed